packages/react-search-ui-views/src/__tests__/SearchBox.test.js

```diff
@@ -60,7 +60,95 @@ it("applies className from inputProps to input element", () => {
     .find("SearchInput")
     .shallow();
   const input = downshift.find(".sui-search-box__text-input");
-  expect(input.props().className).toBe(
-    "sui-search-box__text-input test-class "
-  );
+  expect(input.props().className).toBe("sui-search-box__text-input test-class");
+});
+
+describe("inputView prop", () => {
+  let wrapper, input, button;
+
+  function setup() {
+    wrapper = shallow(
+      <SearchBox
+        {...requiredProps}
+        inputView={({ getAutocomplete, getInputProps, getButtonProps }) => {
+          return (
+            <>
+              <div className="some_custom_wrapper_class">
+                <input
+                  {...getInputProps({
+                    className: "some_custom_input_class"
+                  })}
+                />
+                {getAutocomplete()}
+                <input
+                  {...getButtonProps({
+                    className: "some_custom_button_class"
+                  })}
+                />
+              </div>
+            </>
+          );
+        }}
+      />
+    );
+
+    input = wrapper
+      .dive("Downshift")
+      .find("inputView")
+      .shallow()
+      .find("input")
+      .at(0);
+
+    button = wrapper
+      .dive("Downshift")
+      .find("inputView")
+      .shallow()
+      .find("input")
+      .at(1);
```

Comment:
Good ol' Jest chaining + prettier autolinting :) This is a really clever way of DRYing out the various child diving/shallowing, huge shout out! 💯 

Optional suggestion / feel free to disregard this/ship as-is (like you mentioned, probably good to move on at this point), but if you wanted to be even DRYer and possibly sidestep prettier's linting you could try something like this:

```js
inputView = wrapper
  .dive("Downshift")
  .find("inputView")
  .shallow(); // I think Prettier still multi-line this, oh well
input = inputView.find('input').at(0);
button = inputView.find('button').at(1);
```
and then down on line 113 you can reuse inputView again:
```js
expect(inputView.find('.some_custom_wrapper_class')).toHaveLength(1);
```

Reply:
Good point! Sorry, hit the merge button.